sandplover/plan.py

```diff
@@ -1270,84 +1326,174 @@ def compute_shoreline_roughness(shore_mask, land_mask, **kwargs):
+        which compares all of the "shoreline roughness" metrics implemented
+        in sandplover.
+
     """
-    # extract data from masks
-    if isinstance(land_mask, LandMask):
-        land_mask = land_mask.mask
-        _lm = land_mask.values
-        _dx = float(land_mask[land_mask.dims[0]][1] - land_mask[land_mask.dims[0]][0])
-    elif isinstance(land_mask, xr.core.dataarray.DataArray):
-        _lm = land_mask.values
-        _dx = float(land_mask[land_mask.dims[0]][1] - land_mask[land_mask.dims[0]][0])
-    elif isinstance(land_mask, np.ndarray):
-        _lm = land_mask
+    # process the shore_mask, stripping down to array coordinates regardless of
+    # the input. This simplifies the roughness calculation, because we know
+    # we are dealing with array coords always.
+    if isinstance(shore_mask, ShorelineMask):
+        shore_mask = shore_mask.mask
+        _sm = shore_mask.values
+        _dx = float(
+            shore_mask[shore_mask.dims[0]][1] - shore_mask[shore_mask.dims[0]][0]
+        )
+    elif isinstance(shore_mask, xr.core.dataarray.DataArray):
+        _sm = shore_mask.values
+        _dx = float(
+            shore_mask[shore_mask.dims[0]][1] - shore_mask[shore_mask.dims[0]][0]
+        )
+    elif isinstance(shore_mask, np.ndarray):
+        _sm = shore_mask
         _dx = 1
     else:
-        raise TypeError(f"Invalid type {type(land_mask)}")
+        raise TypeError(f"Invalid type {type(shore_mask)}")
+
+    if calculate_length:
+        _ = kwargs.pop("return_line", None)  # don't allow this to be passed
+        _length = compute_shoreline_length(
+            _sm, return_line=False, **kwargs
+        )  # does not need origin kw, but will take `start`
+    else:
+        _length = np.sum(_sm)
 
-    _ = kwargs.pop("return_line", None)  # trash this variable if passed
-    shorelength = compute_shoreline_length(shore_mask, return_line=False, **kwargs)
+    _area = compute_land_area(land_mask)
 
-    # compute the length of the shoreline and area of land
-    shore_len_pix = shorelength
-    land_area_pix = np.sum(_lm) * _dx * _dx
+    # convert length and area to data coordinates
+    shore_len = _length * _dx
+    land_area = _area  # * _dx * _dx # returned area already in data coordinates!
 
-    if land_area_pix > 0:
-        # compute roughness
-        rough = shore_len_pix / np.sqrt(land_area_pix)
+    if land_area > 0:
+        # compute rugosity
+        rough = shore_len / np.sqrt(land_area)
     else:
-        raise ValueError("No pixels in land mask.")
+        rough = np.nan
+        warnings.warn(
+            "No land area identified in input land_mask, returning np.nan.",
+            category=UserWarning,
+            stacklevel=2,
+        )
 
     return rough
 
 
-def compute_shoreline_rugosity(shore_mask, **kwargs):
-    """Compute shoreline rugosity.
+def compute_shoreline_roughness_variation(shore_mask, origin=(0, 0)):
```

Comment:
I thought we were deprecating origin? I still see it here, would you like to change this?